src/sql/magic.py

```diff
@@ -315,6 +316,25 @@ def interactive_execute_wrapper(**kwargs):
 
         # %%sql {line}
         # {cell}
+
+        # Split the line into tokens separated by whitespace
+        tokens = line.split()
+        arguments = []
+
+        # Iterate through the tokens to separate arguments and SQL code
+        # If the token starts with "--", it is an argument
+        for token in tokens:
+            if token.startswith("--"):
+                arguments.append(token)
+
+        for check_argument in arguments:
+            if check_argument not in _option_strings_from_parser(
+                SqlMagic.execute.parser
+            ):
+                raise exceptions.UsageError(
+                    "Unrecognized argument(s): {}".format(check_argument)
+                )
+
```

Comment:
please abstract this logic into a function and then execute it here

also, since the arguments won't change once the CLI is initialized, you can parse them once (assign the result of `_option_strings_from_parser` into a variable), and then just check that variable, instead of calling `_option_strings_from_parser` every time)

Reply:
Sure, I have updated now